### DISCUSSION TITLE
Write Implied Breadcrumbs to the current file. How should this work?

### DISCUSSION THREAD

**Discussion by SkepticMystic** [SkepticMystic/breadcrumbs#123] [Ideas]
Hey everybody :) I'm working on a new feature which would write yaml to the current file.
For example, if `A` has `parent: [[B]]`, then this feature would write the following to `B` when run:

```
---
child: ['A']
---
```

Note a few things:

1. The syntax of the yaml `relation: ['Link']`. I'm using the syntax that Licat intends to support for links in yaml. This is already picked up by Breadcrumbs, and so I want to encourage its use.
2. It is writing to the yaml frontmatter, even if dataview is enabled. I think this makes the most sense for now. Perhaps in future there could be an option to add it as an inline field instead.
3. Breadcrumbs has to know what the "opposite" field name is. All the information it has is `parent`, so it uses the user's hierarchies to detect that the opposite of that is `child`. If there is no opposite field supplied, it currently uses `<Reverse>Parent`, instead of `child`. This could use some work, so I'm open to suggestions on how to do this better when there is no opposite field to fall back on.

Please share your thoughts on this feature, I will be publishing a working release soon for testing. Let me know if there is something I've missed, or suggestions on how to do this better.

**Comment by verazorita** [comment]
I can see the benefit as if links are to be supported in the metadata this would make a great addon; my only doubt is how kilometric some yaml entries would get.

**Reply by BricoleurSoul** [reply]
> kilometric ... yaml entries 

I decline to admit whether this does, or does not,  reflect on my note naming practices ...

**Reply by SkepticMystic** [reply]
True! Perhaps it would be useful to be able to choose which relations are written to the file.
For example, only write `↑` fields, or only write `↑: Parent` to the file

**Reply by BricoleurSoul** [reply]
Would it be possible to use a comma-delimited field to define where to use?
using your example,  I might want child: ['A']  only to be inserted where the parent  hierarchy value is "complex-system" or where the parent value is "multidisciplinary" but not for any of the other 4 hierarchies I have defined.

**Reply by SkepticMystic** [reply]
Yes, that can definitely be done! Can you make a FR for this specifically :)

**Reply by BricoleurSoul** [reply]
> Yes, that can definitely be done! Can you make a FR for this specifically :)

done.

**Comment by michaelpporter** [comment]
This shipped 🎉 In v4 it's the **"Freeze implied edges to note"** command (Command Palette → Breadcrumbs: Freeze implied edges to note). It writes the note's implied edges into its frontmatter as explicit fields, exactly as proposed here (e.g. an implied `down` becomes a real `down:` property).

Since the feature now exists, closing as resolved.
